dev-tools/verify-fips.bash

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+# This script does some basic sanity checking to verify that the passed binary is FIPS complian
```

Comment:
I'm going to add another e2e test case that does similar checks as this script; it should be enough to make sure that the resulting binary we produce is FIPS compliant/links to OpenSSL.